source/modulo_components/test/cpp/test_component_interface.cpp

```diff
@@ -139,4 +139,18 @@ TYPED_TEST(ComponentInterfaceTest, RaiseError) {
   this->component_->raise_error();
   EXPECT_TRUE(this->component_->get_predicate("in_error_state"));
 }
+
+TYPED_TEST(ComponentInterfaceTest, AddTrigger) {
+  EXPECT_NO_THROW(this->component_->add_trigger("trigger"));
+  ASSERT_FALSE(this->component_->triggers_.find("trigger") == this->component_->triggers_.end());
+  EXPECT_FALSE(this->component_->triggers_.at("trigger"));
+  EXPECT_FALSE(this->component_->get_predicate("trigger"));
+  EXPECT_NO_THROW(this->component_->trigger("trigger"));
+  // After triggering, the trigger will be true only once
+  EXPECT_TRUE(this->component_->triggers_.at("trigger"));
+  EXPECT_TRUE(this->component_->get_predicate("trigger"));
+  // After the predicate function was evaluated once, the trigger is back to false
+  EXPECT_FALSE(this->component_->triggers_.at("trigger"));
+  EXPECT_FALSE(this->component_->get_predicate("trigger"));
```

Comment:
My suggestion about publishing the predicate on triggering will break this test case. Without adding a predicate listener test, you could change it to:

```suggestion
  // When reading, the trigger will be true only once
  this->component_->triggers_.at("trigger") = true;
  EXPECT_TRUE(this->component_->triggers_.at("trigger"));
  EXPECT_TRUE(this->component_->get_predicate("trigger"));
  // After the predicate function was evaluated once, the trigger is back to false
  EXPECT_FALSE(this->component_->triggers_.at("trigger"));
  EXPECT_FALSE(this->component_->get_predicate("trigger"));
```